lib/common/issues/src/dashboard.rs

```diff
@@ -0,0 +1,121 @@
+use std::sync::{Arc, OnceLock};
+
+use dashmap::DashMap;
+
+use crate::issue::{CodeType, Issue, IssueRecord};
+
+#[derive(Default)]
+struct Dashboard {
+    pub issues: DashMap<CodeType, IssueRecord>,
+}
+
+impl Dashboard {
+    /// Activates an issue, returning true if the issue was not active before
+    fn add_issue(&self, issue: impl Issue) -> bool {
+        let code = issue.code();
+        if self.issues.contains_key(&code) {
+            return false;
+        }
+        let issue = IssueRecord::from(issue);
+        self.issues.insert(code, issue).is_none()
+    }
+
+    /// Deactivates an issue by its code, returning true if the issue was active before
+    fn remove_issue<S: AsRef<str>>(&self, code: S) -> bool {
+        if self.issues.contains_key(code.as_ref()) {
+            return self.issues.remove(code.as_ref()).is_some();
```

Comment:
I think you could just do the `remove` operation.
It is most likely optimized will prevent any kind of race condition on our side.

Reply:
Just wanted to make sure it does not request write access without checking with a read access. But it makes sense that it is optimized for these situations. I'll check the source code.

Reply:
After checking the source code for both insert and remove, I found that there is no read before write optimization, see https://docs.rs/dashmap/latest/src/dashmap/lib.rs.html#939

So I'll keep this since it is very probable that issues will want to be re-added or re-removed many times